main.go

```diff
@@ -122,16 +122,19 @@ func main() {
 		os.Exit(1)
 	}
 
-	typeCorrect, err := bootstrap.CheckClusterType(mgr, util.GetMasterNs(mgr.GetAPIReader()))
-	if err != nil {
-		klog.Errorf("Failed to verify cluster type  %v", err)
-		os.Exit(1)
-	}
+	for {
+		typeCorrect, err := bootstrap.CheckClusterType(mgr, util.GetMasterNs(mgr.GetAPIReader()))
+		if err != nil {
+			klog.Errorf("Failed to verify cluster type  %v", err)
+			os.Exit(1)
```

Comment:
@ZhuoxiLi Let's continue the loop if we get the error as well. It will help us reduce the pod crash.